caas/kubernetes/provider/specs/customresourcedefinition.go

```diff
@@ -88,12 +91,6 @@ func UpgradeCustomResourceDefinitionSpecV1Beta1(spec apiextensionsv1beta1.Custom
 			}
 			out.Versions = append(out.Versions, crd)
 		}
-	} else if spec.Version != "" {
-		out.Versions = append(out.Versions, apiextensionsv1.CustomResourceDefinitionVersion{
-			Name:    spec.Version,
-			Served:  true,
-			Storage: true,
```

Comment:
The `apiextensionsv1.CustomResourceDefinitionVersion.Schema` is a required field for `v1`.
Below two cases will fail the k8s API validation:

- the v1beta1 crd spec didn't define `.Validation`
```
creating or
 updating custom resource definitions: ensuring custom resource definition "authcodes.dex.coreos.com" with version "v1beta1": CustomResourceDefinition.apiextensio
ns.k8s.io "authcodes.dex.coreos.com" is invalid: spec.versions[0].schema.openAPIV3Schema: Required value: schemas are required
```

- a valid v1beta1 of `.Validation` might be an invalid v1 `.Schema`

```
creating or
 updating custom resource definitions: ensuring custom resource definition "notebooks.kubeflow.org" with version "v1beta1": CustomResourceDefinition.apiextensions
.k8s.io "notebooks.kubeflow.org" is invalid: [spec.validation.openAPIV3Schema.properties[spec].properties[template].properties[spec].properties[containers].items.
properties[lifecycle].properties[postStart].properties[httpGet].properties[port].anyOf[0].type: Forbidden: must be empty to be structural, spec.validation.openAPI
V3Schema.properties[spec].properties[template].properties[spec].properties[containers].items.properties[lifecycle].properties[postStart].properties[httpGet].prope
rties[port].anyOf[1].type: Forbidden: must be empty to be structural, spec.validation.openAPIV3Schema.properties[spec].properties[template].properties[spec].prope
rties[containers].items.properties[lifecycle].properties[postStart].properties[httpGet].properties[port].type: Required value: must not be empty for specified obj
ect fields, spec.validation.openAPIV3Schema.properties[spec].properties[template].properties[spec].properties[containers].items.properties[lifecycle].properties[p
ostStart].properties[tcpSocket].properties[port].anyOf[0].type: Forbidden: must be empty to be structural, spec.validation.openAPIV3Schema.properties[spec].proper
ties[template].properties[spec].properties[containers].items.properties[lifecycle].properties[postStart].properties[tcpSocket].properties[port].anyOf[1].type: For
bidden: must be empty to be structural, spec.validation.openAPIV3Schema.properties[spec].properties[template].properties[spec].properties[containers].items.proper
ties[lifecycle].properties[postStart].properties[tcpSocket].properties[port].type: Required value: must not be empty for specified object fields, spec.validation.
openAPIV3Schema.properties[spec].properties[template].properties[spec].properties[containers].items.properties[lifecycle].properties[preStop].properties[httpGet].
properties[port].anyOf[0].type: Forbidden: must be empty to be structural, spec.validation.openAPIV3Schema.properties[spec].properties[template].properties[spec].
properties[containers].items.properties[lifecycle].properties[preStop].properties[httpGet].properties[port].anyOf[1].type: Forbidden: must be empty to be structur
al, spec.validation.openAPIV3Schema.properties[spec].properties[template].properties[spec].properties[containers].items.properties[lifecycle].properties[preStop].
properties[httpGet].properties[port].type: Required value: must not be empty for specified object fields, spec.validation.openAPIV3Schema.properties[spec].propert
ies[template].properties[spec].properties[containers].items.properties[lifecycle].properties[preStop].properties[tcpSocket].properties[port].anyOf[0].type: Forbid
den: must be empty to be structural, spec.validation.openAPIV3Schema.properties[spec].properties[template].properties[spec].properties[containers].items.propertie
s[lifecycle].properties[preStop].properties[tcpSocket].properties[port].anyOf[1].type: Forbidden: must be empty to be structural, spec.validation.openAPIV3Schema.
properties[spec].properties[template].properties[spec].properties[containers].items.properties[lifecycle].properties[preStop].properties[tcpSocket].properties[por
t].type: Required value: must not be empty for specified object fields, spec.validation.openAPIV3Schema.properties[spec].properties[template].properties[spec].pro
```

So instead of creating a `.Versions[0]` by Juju itself, we should rely on charmers to upgrade their CRD spec.